src/perf/lib/quicmain.cpp

```diff
@@ -0,0 +1,105 @@
+/*++
+
+    Copyright (c) Microsoft Corporation.
+    Licensed under the MIT License.
+
+Abstract:
+
+    QUIC Perf Main execution engine.
+
+--*/
+
+
+#ifdef QUIC_CLOG
+#include "quicmain.cpp.clog.h"
+#endif
+
+#ifndef _KERNEL_MODE
+#define QUIC_TEST_APIS 1
+#endif
+#include "quic_driver_main.h"
+
+#include "ThroughputServer.h"
+#include "ThroughputClient.h"
+
+const QuicApiTable* MsQuic{nullptr};
+#ifdef _KERNEL_MODE
+uint8_t SelfSignedSecurityHash[20];
+#else
+QUIC_SEC_CONFIG_PARAMS* SelfSignedParams{nullptr};
+#endif
+bool IsSelfSignedValid{ false };
+
+PerfBase* TestToRun{nullptr};
```

Comment:
Whenever possible do not use globals. Especially in general mode, you cannot have a global destructor executed.

Reply:
Destructors won't run, which is why I'm using a pointer and not a unique ptr. I can't realistically pass state back here. In user mode I technically can, but in kernel mode I'd just have the same thing 1 layer up.

Also, MsQuic HAS to be global because of all the headers that expect it to be global.

Reply:
I agree some pointers need to be global. I'm just saying whenever possible do not have a global.